Tangem/App/Utilities/SearchUtil.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  SearchUtil.swift
+//  TangemApp
+//
+//  Created by Aleksei Muraveinik on 3.11.24..
+//  Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+enum SearchUtil<T> {
+    static func search(_ items: [T], in keyPath: KeyPath<T, String>, for searchText: String) -> [T] {
+        if searchText.isEmpty {
+            return items
+        }
+
+        let loweccasedSearchText = searchText.lowercased()
```

Comment:
может еще `trimmed` ?
```suggestion
        let loweccasedSearchText = searchText.lowercased().trimmed()
```